### PR TITLE
[generate client] Add post processing option when generating from autorest

### PR DESCRIPTION
**Pull request by catalinaperalta**

### CODE REVIEW

File: tools/azure-sdk-tools/packaging_tools/generate_client.py [hunk 1]
```diff
@@ -2,11 +2,28 @@
...
 
+CONF_NAME = "pyproject.toml"
+
+def check_post_process(folder: Path) -> bool:
+    conf_path = folder / CONF_NAME
+    if not conf_path.exists():
+        return False
+
+    with open(conf_path, "rb") as fd:
+        return tomli.load(fd)["generate"]["autorest-post-process"]
```
Comment (catalinaperalta) [inline tools/azure-sdk-tools/packaging_tools/generate_client.py:18]: What's the best place to document this for folks to use?

Comment (catalinaperalta) [reply tools/azure-sdk-tools/packaging_tools/generate_client.py:18]: Adding a check here if there's not generate config in the toml

File: tools/azure-sdk-tools/packaging_tools/generate_client.py [hunk 2]
```diff
@@ -2,11 +2,31 @@
...
+
+def check_post_process(folder: Path) -> bool:
+    conf_path = folder / CONF_NAME
+    if not conf_path.exists():
+        return False
+
+    with open(conf_path, "rb") as fd:
+        toml_dict = tomli.load(fd)
+        if toml_dict.get("tox-generate", None) is not None:
+            return toml_dict["tox-generate"]["autorest-post-process"]
```
Comment (kristapratico) [inline tools/azure-sdk-tools/packaging_tools/generate_client.py:20]: I think we need to adjust how we're accessing the dict. If I read the pyproject.toml from, for example, textanalytics, I get a dict that looks like this:

```
>>> import tomli
>>> with open("pyproject.toml", "rb") as fd:
...     toml_dict = tomli.load(fd)
...
>>> toml_dict
{'tool': {'azure-sdk-build': {'pyright': False}}}
```

We have a section for azure-sdk-build checks, do we want to add one for generation options?

Comment (catalinaperalta) [reply tools/azure-sdk-tools/packaging_tools/generate_client.py:20]: something like `[tools.generate]`? I think Laurent originally suggested that, I'm good updating to use that

Comment (catalinaperalta) [reply tools/azure-sdk-tools/packaging_tools/generate_client.py:20]: Changed it so the section is `[tool.generate]`

File: tools/azure-sdk-tools/packaging_tools/generate_client.py (hunk 1 above)
Comment (kristapratico) [reply tools/azure-sdk-tools/packaging_tools/generate_client.py]: > What's the best place to document this for folks to use?

I was going to suggest eng_sys_checks.md, but that's really focused on how to manage the CI checks. Maybe somewhere where we talk about generating python libraries? Here's the wiki for [swagger](https://github.com/Azure/azure-sdk-for-python/wiki/Dataplane-Codegen-Quick-Start-For-Swagger), not sure if we have one for typespec

**Review by kristapratico — APPROVED** [review]

File: tools/azure-sdk-tools/packaging_tools/generate_client.py (hunk 1 above)
Comment (catalinaperalta) [reply tools/azure-sdk-tools/packaging_tools/generate_client.py]: The default quick start describes how to generate with TypeSpec, but I think adding this in the swagger wiki makes the most sense. Thanks! :)

**Pull request merged**